crates/bevy_ecs/src/system/query.rs

```diff
@@ -212,6 +212,42 @@ where
             Err(QueryComponentError::MissingWriteAccess)
         }
     }
+
+    pub fn get_unique(&self) -> Result<<Q::Fetch as Fetch<'_>>::Item, UniqueQueryError<'_, Q>>
```

Comment:
You may want to box the error to shrink the return type.

Reply:
How would that help?

Reply:
This could improve performance if the smaller return type fits into two registers by avoiding stack accesses.

Reply:
Hmm, I see... Personally I don't think this would change the performance much, but I don't have a way to benchmark that. This seems quite micro-optimization-y, so I'd rather wait on this until we have realistic uses of this function we can benchmark.

Reply:
Maybe we don't need to return `Item` at all in the Error? The query isn't consumed by this operation, so users that want the other values in the event of an error can always iterate over them. Generally when someone wants a "single" item and there is more than one, there is nothing particularly special about the first item.

Reply:
I think we should resolve this conversation before merging. My preference is to remove "result: Item" from Error. What do you think @TheRawMeatball ?

Reply:
Hmm, while I prefer it as-is I suppose the use case is very slim and I'm willing to compromise to get the ball rolling. I'll update the PR to remove this now.

Reply:
Thanks!